compiler/rustc_expand/src/mbe/metavar_expr.rs

```diff
@@ -10,6 +10,10 @@ use rustc_span::Span;
 /// A meta-variable expression, for expansions based on properties of meta-variables.
 #[derive(Debug, Clone, PartialEq, Encodable, Decodable)]
 pub(crate) enum MetaVarExpr {
+    /// Unification of two identifiers. The `bool` of each element indicates if there is a
+    /// preceding dollar sign.
+    Concat { lhs_ident: Ident, lhs_is_var: bool, rhs_ident: Ident, rhs_is_var: bool },
```

Comment:
I think we may want to support concat for multiple items, i.e. make this a `Box<[T]>`. 

It would also be nice to have [case conversion support](https://docs.rs/paste/latest/paste/#case-conversion) so this can drop in place of `paste!`, but that can come later

Reply:
Sure, the introduction of many identifiers is a generalization of the current behavior and is very handy.